script.js

```diff
@@ -17,44 +16,85 @@ const render = () => {
   const p = document.querySelector(".sum");
   allFinances.forEach((el, idx) => {
-    sum += el.money;
+    sum += Number(el.money);
+    const updateEl = {};
+    updateEl._id = el._id;
 
     const contentBox = document.createElement("div");
     contentBox.className = "contentBox";
 
     const nameCompany = document.createElement("p");
-    nameCompany.className = "nameCompany";
+    nameCompany.className = el.isEdit ? "displayNone" : "nameCompany";
     nameCompany.textContent = `${idx + 1}) ${el.nameCompany}`;
     contentBox.appendChild(nameCompany);
 
+    const editInputName = document.createElement("input");
+    editInputName.className = el.isEdit ? "editInputName" : "displayNone";
+    editInputName.value = el.nameCompany;
+    updateEl.nameCompany = el.nameCompany;
+    editInputName.addEventListener("change", (e) => {
+      updateEl.nameCompany = e.target.value.trim();
+    });
+    contentBox.appendChild(editInputName);
+
     const date = document.createElement("p");
+    date.className = el.isEdit ? "displayNone" : "";
     date.textContent = convertDate(el.date);
     contentBox.appendChild(date);
 
+    const editInputDate = document.createElement("input");
+    editInputDate.className = el.isEdit ? "editInputDate" : "displayNone";
+    editInputDate.type = "date";
+    editInputDate.value = el.date.slice(0, 10);
+    updateEl.date = el.date.slice(0, 10);
+    editInputDate.addEventListener("change", (e) => {
+      updateEl.date = e.target.value;
+    });
+    contentBox.appendChild(editInputDate);
+
     const money = document.createElement("p");
+    money.className = el.isEdit ? "displayNone" : "";
     money.textContent = `${el.money} p.`;
     contentBox.appendChild(money);
 
+    const editInputMoney = document.createElement("input");
+    editInputMoney.className = el.isEdit ? "editInputMoney" : "displayNone";
+    editInputMoney.type = "number";
+    editInputMoney.value = el.money;
+    updateEl.money = el.money;
+    editInputMoney.addEventListener("change", (e) => {
+      updateEl.money = Number(e.target.value);
+    });
+    contentBox.appendChild(editInputMoney);
+
     const iconBox = document.createElement("div");
     iconBox.className = "iconBox";
-
     const editIcon = document.createElement("img");
     editIcon.src = "image/pen.svg";
+    editIcon.className = el.isEdit ? "displayNone" : "";
+    editIcon.addEventListener("click", () => onClickEdit(idx));
     iconBox.appendChild(editIcon);
 
+    const doneIcon = document.createElement("img");
+    doneIcon.src = "image/check.svg";
+    doneIcon.className = el.isEdit ? "" : "displayNone";
+    doneIcon.addEventListener("click", () => onClickSaveEl(updateEl, idx));
+    iconBox.appendChild(doneIcon);
+
     const deleteIcon = document.createElement("img");
     deleteIcon.src = "image/trash.svg";
-    deleteIcon.addEventListener("click", () => onClickDelete(idx));
+    el.isEdit ? deleteIcon.addEventListener("click", () => onClickClose(idx)) :
```

Comment:
use ternary operator in choose functions
deleteIcon.addEventListener("click", () => el.isEdit ? onClickClose(idx) : onClickDelete(idx));